src/reconcile/rec_write.c

```diff
@@ -2290,19 +2289,19 @@ __rec_hs_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r)
     if (i == r->multi_next)
         return (0);
 
-    __wt_hs_cursor(session, &cursor, &session_flags);
+    WT_RET(__wt_hs_cursor(session, &session_flags));
 
     for (multi = r->multi, i = 0; i < r->multi_next; ++multi, ++i)
         if (multi->supd != NULL) {
-            WT_ERR(__wt_hs_insert_updates(cursor, S2BT(session), r, multi));
+            WT_ERR(__wt_hs_insert_updates(session->hs_cursor, S2BT(session), r, multi));
             if (!r->leave_dirty) {
                 __wt_free(session, multi->supd);
                 multi->supd_entries = 0;
             }
         }
 
 err:
-    WT_TRET(__wt_hs_cursor_close(session, &cursor, session_flags));
+    WT_TRET(__wt_hs_cursor_close(session, session_flags));
```

Comment:
Doesn't this actually close the cursor? I thought the point of leaving it on the session is so that you just leave it open and close it when the session closes (getting the same effect as the pool).

Reply:
We want to manage the HS cursor lifetime using the default cursor caching mechanism. We can effectively call a close on the cursor and rely on cursor caching to keep it open for a certain duration. WT-5477 which duplicates to WT-5501 is going to resolve cursor caching issues.